File: src/main/java/net/kenro/ji/jin/purescript/psi/impl/PSProgramImpl.java
package net.kenro.ji.jin.purescript.psi.impl;

import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiElementVisitor;
import net.kenro.ji.jin.purescript.psi.PSProgram;
import net.kenro.ji.jin.purescript.psi.PSVisitor;
import org.jetbrains.annotations.NotNull;

public class PSProgramImpl extends PSPsiElement implements PSProgram {

    public PSProgramImpl(final ASTNode node) {
        super(node);
    }

}

File: src/main/java/net/kenro/ji/jin/purescript/psi/impl/PSProgramImpl.kt
package net.kenro.ji.jin.purescript.psi.impl

import com.intellij.lang.ASTNode
import net.kenro.ji.jin.purescript.psi.PSProgram

class PSProgramImpl(node: ASTNode) : PSPsiElement(node), PSProgram {
    val module: PSModuleImpl
        get() = this.firstChild as PSModuleImpl
}